giddy/markov.py

```diff
@@ -615,6 +619,20 @@ def __init__(self, y, w, k=4, m=4, permutations=0, fixed=True,
                 y_int.append(list(map(label_dict.get, yi)))
             self.class_ids = np.array(y_int)
             self.lclass_ids = self.class_ids
+        elif manual_lag is not False: 
+            assert len(self.manual_lag) == len(y), "`lags must be the same length as the observation data"
+            merged = list(itertools.chain.from_iterable(y))
+            classes = np.unique(merged)
+            self.classes = classes
+            self.k = len(classes)
+            self.m = self.k
+            label_dict = dict(zip(classes, range(self.k)))
+            y_int = []
```

Comment:
Since this option also requires the input time series `y` to be discrete, and thus there is much duplicate code with when `discrete=True`, I think it makes sense to subsume the option of `manual_lag` within  the code block `if discrete=True` to avoid the duplication?

```python
if discrete:
    merged = list(itertools.chain.from_iterable(y))
    classes = np.unique(merged)
    self.classes = classes
     ...
    if manual_lag:
        ...

```

Reply:
yeah that probably makes more sense